integration_tests/chain.go

```diff
@@ -0,0 +1,299 @@
+package integration_tests
+
+import (
+	"fmt"
+	"github.com/peggyjv/gravity-bridge/module/app"
+	"github.com/peggyjv/gravity-bridge/module/app/params"
+	"io/ioutil"
+	"os"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdkTypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	sdkTx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	gravitytypes "github.com/peggyjv/gravity-bridge/module/x/gravity/types"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+	rpchttp "github.com/tendermint/tendermint/rpc/client/http"
+)
+
+const (
+	keyringPassphrase = "testpassphrase"
+	keyringAppName    = "testnet"
+)
+
+var (
+	encodingConfig params.EncodingConfig
+	cdc            codec.Codec
+)
+
+func init() {
+	encodingConfig = app.MakeEncodingConfig()
+
+	encodingConfig.InterfaceRegistry.RegisterImplementations(
```

Comment:
Shouldn't the app encoding config have this stuff in it?

Reply:
This is for the test, which I don't believe pulls in app.go, we explicitly choose the ones we will use

Reply:
Or maybe it does, and I added this before checking?